routes/admin.js

```diff
@@ -68,6 +80,121 @@ router.get('/', ensureEditorOrAdmin, function(req, res) {
 	});
 });
 
+/////////// LOCATIONS ///////////
+
+router.get('/locations', ensureEditorOrAdmin, function(req, res) {
+	var models = req.app.get('models');
+
+	models.Location.findAll({
+		limit: 20,
+	}).then(function(locations) {
+		res.render('locations', {
+			user: req.user,
+			locations: locations
+		});
+	});
+});
+
+router.get('/location/:location_id', ensureEditorOrAdmin, function(req, res) {
```

Comment:
For me, this route needs to appear after the other `/location` ones, otherwise I get a 404 error when express tries to load up the page for a location with `id = "add"`. Does it work for you?